felix/bpf/maps/maps.go

```diff
@@ -757,6 +785,12 @@ func (b *PinnedMap) CopyDeltaFromOldMap() error {
 	if err != nil {
 		return fmt.Errorf("error upgrading data from old map %s, err=%w", b.GetName(), err)
 	}
+
+	err = b.DeletePreviousVersion()
+	if err != nil {
+		return fmt.Errorf("failed to delete previous %s map, err=%w", b.Name, err)
```

Comment:
I think we should check whether the failure was because the map did not exist essentially getting ENOENT. In that case, that is not quite an error as we are in the state where we want to be and thus we can continue. 

Note that this function is called on for CT map https://github.com/projectcalico/calico/blob/2ebc1f92ecc39332cf1d55ba676d9101af24982f/felix/dataplane/linux/bpf_ep_mgr.go#L1341

But we need to make sure that any old map is deleted. For instance, state maps are not being upgraded as they are internal to the version of the programs that use them, but they are pinned so when we upgrade, they may be left over from the previous versions.

CT is sort of special, because we need to move the state from one version to the next verion so that we do not break connections. That is not true for most of the other maps.

Reply:
Thanks @tomastigera - actually I see a neat helper function `IsNotExists()` which wraps this check for ENOENT so I have implemented as per comment following an existing example for consistency

Reply:
but as I said :arrow_up: this only fixes the issue for CT map. There are many more. So we need to tke it out of the copy-deltas and add a test that also mkes sure that other aps are removed. The per-cpu state map ay be a good example.

Reply:
All of the above address in PR 8012 - therefore will close this PR now - thanks